src/main/kotlin/com/github/vol0n/utbotcppclion/actions/GenerateForPredicateAction.kt

```diff
@@ -0,0 +1,246 @@
+package com.github.vol0n.utbotcppclion.actions
+
+import com.github.vol0n.utbotcppclion.actions.utils.client
+import com.github.vol0n.utbotcppclion.actions.utils.coroutinesScopeForGrpc
+import com.github.vol0n.utbotcppclion.actions.utils.getContainingFunFromAction
+import com.github.vol0n.utbotcppclion.ui.GeneratorSettingsDialog
+import com.github.vol0n.utbotcppclion.utils.handleTestsResponse
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.ui.ComponentValidator
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.ui.popup.JBPopup
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.ui.DocumentAdapter
+import com.intellij.ui.components.fields.ExtendableTextField
+import javax.swing.ListSelectionModel
+import javax.swing.event.DocumentEvent
+import testsgen.Util.ValidationType
+import java.awt.Dimension
+import java.awt.event.KeyAdapter
+import java.awt.event.KeyEvent
+import java.math.BigInteger
+import java.util.function.Supplier
+import kotlinx.coroutines.launch
+import testsgen.Testgen
+
+class GenerateForPredicateAction: AnAction() {
+
+    override fun update(e: AnActionEvent) {
+        val containingFun = getContainingFunFromAction(e)
+        e.presentation.isEnabledAndVisible = (containingFun != null)
+    }
+
+    fun createListPopup(title: String, list: List<String>, onChoose: (String) -> Unit): JBPopup {
+        return JBPopupFactory.getInstance().createPopupChooserBuilder(list)
+            .setResizable(false)
+            .setMovable(false)
+            .setTitle(title)
+            .setSelectionMode(ListSelectionModel.SINGLE_SELECTION)
+            .setCloseOnEnter(true)
+            .setItemChosenCallback(onChoose)
+            .createPopup()
+    }
+
+    override fun actionPerformed(e: AnActionEvent) {
+
+        fun sendPredicateToServer(validationType: ValidationType, returnValue: String, predicate: String) {
```

Comment:
You have all buildFileRequestFromEvent and other functions in utils, but this not in utils. Please make uniform so that either all in utils or all in their separate *Action.kt